### ISSUE TITLE
Give mark button a title attribute / popup thingy

### ISSUE THREAD

**Issue by jdevalk** [Yoast/YoastSEO.js#623]
Some text to indicate what pressing this button would do.

Paging @afercia  :)

**Comment by afercia** [comment]
:) yep icon-only controls would need some kind of "label" or text to reduce ambiguity. Some background on this very interesting article: https://www.nngroup.com/articles/icon-usability/ The Norman group folks are known to be a bit "strict" and they suggest visible text close to icons. I share the feeling of that article and there's one sentence I'd like to highlight:
> if it takes you more than 5 seconds to think of an appropriate icon for something, it is unlikely that an icon can effectively communicate that meaning

By the way, we need a pragmatical approach and I think there are basically two options: title attributes or "accessible tooltips". The latter is also a topic I had the opportunity to discuss with other people in the WP a11y team.

General considerations about standard title attributes:
- they work just on hover
- on focus, nothing happens
- useless on mobile and touch devices
- current trend in WordPress is to remove all the title attributes

General considerations about "tooltips"
- it really all depends on how they're built :)

Many big applications make use of "tooltips" for icons. Thinking at Gmail, Twitter, Facebook, and even GitHub. Gmail uses tooltips for most of the controls that don't have visible text. They work both on hover and focus but it's a JS based implementation and pretty un-semantic.

Facebook, when i checked some months ago, used tooltips getting the text from their hidden accessible text elements, see below (but it's a JS based implementation):

![tooltips facebook aprile 2016](https://cloud.githubusercontent.com/assets/1682452/15859447/7f9f5480-2cc5-11e6-8f20-a844ee165930.png)

GitHub: I like their implementation because it's CSS based and it's accessible. It gets the content from the `aria-label`  attribute and uses that with CSS pseudo elements to build the tooltip:

<img width="200" alt="gh notifications icon" src="https://cloud.githubusercontent.com/assets/1682452/15859553/e4b958c0-2cc5-11e6-8ad9-d384d968a466.png">

- the `aria-label` covers part of the accessibility issues
- pseudo elements are just CSS, they work on any controls "state" (hover, focus, active)
- light and works even without JS support

I'd vote for a solution similar to the GitHub implementation. Couple of considerations:
It's sort of a big UI shift, maybe something to consider for general use, not limited to just one single scenario. Also something to consider wisely because once tooltips get introduced, it would be hard to remove them. Worth noting at the moment the only implementation of tooltips in WordPress is in TinyMCE:
<img width="181" alt="screen shot 2016-06-07 at 14 59 49" src="https://cloud.githubusercontent.com/assets/1682452/15860027/bb99eaca-2cc7-11e6-914f-6109d42fb81f.png">

Thinking at touch devices, we should consider if it's worth trying to find a way to reveal the tooltips after a "longpress" event. Maybe something to consider for the future.

**Comment by afercia** [comment]
Editing in-browser and just copying and pasting the GH style without any adjustment:

<img width="828" alt="tooltipped" src="https://cloud.githubusercontent.com/assets/1682452/15861218/2eb809de-2ccc-11e6-8e59-86485a3444ad.png">

**Comment by afercia** [comment]
Opened a [new issue on wordpress-seo](https://github.com/Yoast/wordpress-seo/issues/4804) since we're considering to introduce tooltips as a general UI tool.

**Comment by afercia** [comment]
Testing the tooltips see https://github.com/Yoast/wordpress-seo/issues/4804 on the mark buttons with their new position on the left:

on hover:

<img width="921" alt="tooltip hover" src="https://cloud.githubusercontent.com/assets/1682452/15925258/b20be4a2-2e36-11e6-9134-2eb34d6ceed9.png">

on focus:

<img width="921" alt="tooltip focus" src="https://cloud.githubusercontent.com/assets/1682452/15925244/a5a3f0b0-2e36-11e6-86d2-0bfd5745695c.png">

Using VoiceOver:

<img width="923" alt="screen shot 2016-06-09 at 11 33 38" src="https://cloud.githubusercontent.com/assets/1682452/15925276/c569cdde-2e36-11e6-8a41-c734f218d459.png">

**Comment by jdevalk** [comment]
Looks good @afercia !! Can / should we use the same for our score bullets?

**Comment by afercia** [comment]
@jdevalk we can try but on the score icons tooltips would shop up only on hover since they're not focusable. I'm a bit concerned this could introduce a bit of confusion about what is an actionable control and what is not.

**Comment by rensw90** [comment]
CR-Done, looks good 👍

**Comment by jdevalk** [comment]
@afercia good point, let's not do that then.
